lib/galaxy_test/api/test_tools.py

```diff
@@ -433,6 +433,70 @@ def test_test_data_download_composite(self):
         expected_names = {'velveth_test1/Roadmaps', 'velveth_test1/output.html', 'velveth_test1/Sequences', 'velveth_test1/Log', 'velveth_test1/output/', 'velveth_test1/output/1'}
         assert set(namelist) == expected_names
 
+    @uses_test_history(require_new=False)
+    def test_convert_dataset_explicit_history(self, history_id):
+        fasta1_contents = open(self.get_filename("1.fasta")).read()
+        hda1 = self.dataset_populator.new_dataset(history_id, content=fasta1_contents)
+
+        payload = {
+            "src": "hda",
+            "id": hda1["id"],
+            "source_datatype": "fasta",
+            "target_datatype": "tabular",
+            "history_id": history_id
+        }
+        create_response = self._post("tools/CONVERTER_fasta_to_tabular/convert", data=payload)
+        create_response.raise_for_status()
+        assert create_response.json()["outputs"][0]["file_ext"] == "tabular"
+
+    @uses_test_history(require_new=False)
+    def test_convert_dataset_implicit_history(self, history_id):
+        fasta1_contents = open(self.get_filename("1.fasta")).read()
+        hda1 = self.dataset_populator.new_dataset(history_id, content=fasta1_contents)
+        payload = {
+            "src": "hda",
+            "id": hda1["id"],
+            "source_datatype": "fasta",
+            "target_datatype": "tabular"
+        }
+        create_response = self._post("tools/CONVERTER_fasta_to_tabular/convert", data=payload)
+        create_response.raise_for_status()
+        assert create_response.json()["outputs"][0]["file_ext"] == "tabular"
+
+    @uses_test_history(require_new=False)
+    def test_convert_hdca(self, history_id):
+        data = [
+            {
+                "name": "test0",
+                "elements": [
+                    {"src": "pasted", "paste_content": "123\n", "name": "forward", "ext": "fasta"},
+                    {"src": "pasted", "paste_content": "456\n", "name": "reverse", "ext": "fasta"},
+                ]
+            },
+            {
+                "name": "test1",
+                "elements": [
+                    {"src": "pasted", "paste_content": "789\n", "name": "forward", "ext": "fasta"},
+                    {"src": "pasted", "paste_content": "0ab\n", "name": "reverse", "ext": "fasta"},
+                ]
+            }
+        ]
+        hdca1 = self.dataset_collection_populator.upload_collection(history_id, "list:paired", elements=data)
+        self._assert_status_code_is(hdca1, 200)
+
+        payload = {
+            "src": "hdca",
+            "id": hdca1.json()["outputs"][0]["id"],
+            "source_datatype": "fasta",
+            "target_datatype": "tabular",
+            "history_id": history_id
+        }
+        create_response = self._post("tools/CONVERTER_fasta_to_tabular/convert", payload)
+        create_response.raise_for_status()
+        assert create_response.json()["outputs"][0]["file_ext"] == "tabular"
```

Comment:
Will you please verify ``create_response.json()["implicit_collections"]`` is non-empty here and is empty in the other tests.

I really wanted these tests to be a bit more flushed out - you should wait on the job (there are abstractions for doing this and the job id is in that response object) and then re-fetch the outputs and make sure they are correct. The implicit collection above should have all its contents have the correct file_ext.